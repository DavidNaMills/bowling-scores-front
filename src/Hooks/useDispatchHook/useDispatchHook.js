import {useDispatch } from 'react-redux';
import * as actions from '../../store/allActions';


const useDispatchHook = () =>{
    const dispatch = useDispatch();

    //Game actions
    const addNewPlayerDispatch = (data) => dispatch(actions.addPlayer(data));
    const addNewGameDispatch = (data) => dispatch(actions.addNewGame(data));
    const initGameDispatch = (data) => dispatch(actions.initGame(data));
    const commitNewGameDispatch = (data) => dispatch(actions.commitNewGame(data));

    const removePlayerDispatch = (id) => dispatch(actions.removePlayer(id));
    const updateScoresDispatch = (data) => dispatch(actions.updateIndividualScore(data))

    //User actions
    const loginUserDispatch = (data) => dispatch(actions.userLogin(data));
    const logoutUserDispatch = () => dispatch(actions.userLogout())

    return {
        addNewPlayerDispatch,
        addNewGameDispatch,
        initGameDispatch,
        commitNewGameDispatch,
        
        removePlayerDispatch,
        updateScoresDispatch,

        loginUserDispatch,
        logoutUserDispatch
    }
}

export default useDispatchHook;