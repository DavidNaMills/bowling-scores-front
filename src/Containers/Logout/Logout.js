import React, {useEffect} from 'react';
import {Redirect} from 'react-router-dom';
import useDispatchHook from '../../Hooks/useDispatchHook/useDispatchHook';

const Logout = () =>{
    const {logoutUserDispatch} = useDispatch();

    useEffect(()=>{
        logoutUserDispatch();
    }, []);

    return (
        <Redirect to='/'/>
    )
}

export default Logout;