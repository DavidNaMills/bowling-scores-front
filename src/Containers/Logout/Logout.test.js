import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {Redirect, BrowserRouter} from 'react-router-dom';
import Logout from './Logout';

jest.mock('../../Hooks/useDispatchHook/useDispatchHook');
import useDispatchHook from '../../Hooks/useDispatchHook/useDispatchHook';
import { useDispatch } from 'react-redux';

const mockLogoutDispatch = jest.fn();
useDispatchHook.mockImplementation(()=>({
    logoutUserDispatch: mockLogoutDispatch
}));

afterEach(()=>{
    mockLogoutDispatch.mockClear();
});

describe('<Logout/> test suite', ()=>{
    it('snaphot test', ()=>{
        const component = renderer.create(<BrowserRouter><Logout /></BrowserRouter>);
        const app = component.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('renders a <Redirect/> component', ()=>{
        const wrapper = mount(<BrowserRouter><Logout /></BrowserRouter>);
        expect(wrapper.find(Redirect).length).toBe(1);
        expect(mockLogoutDispatch).toHaveBeenCalledTimes(1);
    });
});