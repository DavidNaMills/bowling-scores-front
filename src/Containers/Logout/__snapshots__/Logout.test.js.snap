// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout/> test suite snaphot test 1`] = `null`;
